GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -998,14 +1001,25 @@ private bool UnStageSelectedFile()
         /// <returns>true if executed</returns>
         private bool ResetSelectedFiles()
         {
-            // No Reset in DiffFiles - unclear which rev to reset to
-            if (!DiffText.ContainsFocus || !DiffText.SupportLinePatching)
+            if (DiffText.ContainsFocus && DiffText.SupportLinePatching)
             {
-                return false;
+                ResetOrRevertSelectedLines();
+                return true;
+            }
+            else if (DiffFiles.Focused)
+            {
+                var parents = DiffFiles.SelectedItemParents.ToList();
+                if (parents.Count != 1 || !CanResetToRevision(parents.FirstOrDefault()))
+                {
+                    return false;
+                }
+
+                // Reset to first (parent)
+                ResetSelectedItemsTo(false);
+                return true;
```

Comment:
readability can be improved
```suggestion
                if (parents.Count == 1 && CanResetToRevision(parents.FirstOrDefault()))
                {
                    // Reset to first (parent)
                    ResetSelectedItemsTo(actsAsChild: false);
                    return true;
                }
```

Reply:
I can agree on the readability and it follows the pattern better, but @RussKie usually requests as early exits as possible...

Reply:
will change, calculate revisions before first if to reduce depth (readability is more important than a few CPU cycles saved when line patching)